app/misc/authenticate.ts

```diff
@@ -80,6 +80,12 @@ function ModalSignIn(callback) {
         getUserInfo(callback);
 }
 
+// provide a fresh cred object. 
+// previously we were trying to use same object again and again which was cauing issues
```

Comment:
do we know why it was causing issues?

Reply:
found nothing from any changelog, but did found similar issues (4-5) many of them were unsolved (i did post my solution there too). one of them had the solution similar to what I came up with too. 
It looks like push/pull code might be calling some cred cleanup, and reusing the cred object is not possible. btw in the terminal also we have to enter login details every time right?

Reply:
how to find out why? I mean it was low-level segmentation fault error (assertion fail). similar error u can produce by unplugging ur internet and press pull. (also fixed) 
I mean I don't know how to go below to c level debugging in this application.

Reply:
> found nothing from any changelog, but did found similar issues (4-5) many of them were unsolved (i did post my solution there too). one of them had the solution similar to what I came up with too.
> It looks like push/pull code might be calling some cred cleanup, and reusing the cred object is not possible. btw in the terminal also we have to enter login details every time right?

It depends on your env setup. That's fine, I was just wondering if we knew the underlying issue. And you do!

I would suggest adding your comments here in the code -- perhaps with a reference to an open issue with nodegit?